company/okrs/2020_q1.md

```diff
@@ -0,0 +1,105 @@
+   1. VP Eng KR: Each team completes >80% of the work that they committed to at the beginning of the release cycle each iteration.
+
+1. **Distribution: Resolve issues for critical customer deals and expansions**
+1. **Distribution: Improve developer productivity**
+   1. Distribution KR: Lower build times to X
+   1. Distribution KR: Automate away release captain role
+   1. Distribution KR: Infrastructure for automated deploys and releases (no human in loop required by default)
+1. **Distribution: Build features that accelerate customers through pipeline**
+   1. Distribution KR: Shorten the time from "intent to try Sourcegraph" to "whole org is using Sourcegraph" (onboarding experience)
+   1. Distribution KR: Ensure new features (LSIF, GitLab Native Integration, Automation) are accessible and easily tryable by customers once they go GA
+   1. Distribution KR: Surface information to site admins and managers to encourage expansion/upsell
+
+1. **Code intelligence: LSIF is adopted by customers**
+   1. Code intelligence KR: At least three customers are using LSIF on their primary repositories
+   1. Code intelligence KR: At least one customer is using LSIF across all of their code in languages for which there exists an LSIF indexer
+   1. Code intelligence KR: Start tracking the performance (user interaction to result) of hover tooltips, go to definition, and find references
+
+1. **Code intelligence: Increase user confidence in code intelligence provided by LSIF**
+   1. Code intelligence KR: Code intelligence UI (tooltips, references panel, etc.) indicates whether or not results are precise
+   1. Code intelligence KR: Fall back to fuzzy if the file has been modified since the last LSIF upload
+   1. Code intelligence KR: Make use of LSIF data when the current line is not part of the diff since the last LSIF upload
+
+1. **Code intelligence: Make search better using precise code intel data**
+   1. Code intelligence KR: Implement search within dependencies and/or dependents of current repo
+   1. Code intelligence KR: Implement searching for defs/refs by symbol name
+
+1. **Web: Make users aware of advanced search features**
+   1. Web KR: Start tracking time to first search and search filters per query in pings
+   1. Web KR: Shorten time to first search
+   1. Web KR: Increase average number of search filters per query
+
+1. **Web: Automation is adopted by customers**
+   1. Web KR: At least four customers are using campaigns in production
+   1. Web KR: All campaign types have been used by customers in production
+
+1. **Core Services: Search gains new capabilities which are adopted by customers**
+   1. Core Services KR: Support AND/OR/NOT operators on file contents.
+   1. Core Services KR: Write at least 2 pieces of technical documentation (docs, walkthroughs, blog posts) about search functionality.
+   1. Core Services KR: Structural search can be ran at scale (~30k repositories).
```

Comment:
```suggestion
   1. Core Services KR: Ensure time-to-first-result for search requests in the 99th percentile latency of 2s, up to any number of repositories.
```

@tsenart are your thoughts to have this for 'search' generally, or structural search only? I think with the addition of AND/OR/NOT which will demand more compute or multiple search passes, it is worth adding this cap this for any search request. Also, since basically my first day at Sourcegraph I've been unhappy that it takes >2s to list all repos for a simple query like `repo:github.com/rvantonder` or `repo:github.com/tsenart` on sourcegraph.com [and it still takes >2s](https://sourcegraph.com/search?q=repo:github.com/rvantonder&patternType=literal) so I really think the other forms of search can use some attention.

Reply:
> are your thoughts to have this for 'search' generally, or structural search only?

Search, generally.